core/src/main/scala/kafka/controller/ControllerEventManager.scala

```diff
@@ -93,9 +93,14 @@ class ControllerEventManager(controllerId: Int,
   def start(): Unit = thread.start()
 
   def close(): Unit = {
-    thread.initiateShutdown()
-    clearAndPut(ShutdownEventThread)
-    thread.awaitShutdown()
+    try {
+      thread.initiateShutdown()
+      clearAndPut(ShutdownEventThread)
+      thread.awaitShutdown()
+    } finally {
+      removeMetric("EventQueueSize")
+      removeMetric("EventQueueTimeMs")
```

Comment:
Should we create a constant for these metric names?

Reply:
Yep, may as well.